### PR TITLE
feat(bulk): replace MTGJSON with Scryfall bulk data

### PR DESCRIPTION
**Pull request by j4th**
## Summary

- **Replace MTGJSON with Scryfall Oracle Cards bulk data** — richer card info (prices, legalities, images, EDHREC rank) with rate-limit-free O(1) lookups
- **Filter non-playable layouts** (minigames, art series, tokens, emblems) that were silently overwriting real cards in bulk data
- **Add tiered test strategy** — integration tests (fixture-mocked E2E), live smoke tests (real server + real APIs), and `check:quick` fast gate
- **Add Phase 6 design specs** for infrastructure modernization, cross-format tools, and format workflows
- **Add interactive smoke test skill** for end-to-end MCP server verification

### New: Scryfall bulk data service + provider
- `ScryfallBulkClient` with lazy download, ETag-based conditional refresh, background refresh, layout filtering
- `bulk_card_lookup` and `bulk_card_search` tools (namespace: `bulk_`)
- Returns full `Card` objects matching Scryfall API shape
- 789-line test suite with adversarial fixtures (minigames, art series, tokens, emblems)

### Removed: MTGJSON
- Deleted `services/mtgjson.py`, `providers/mtgjson.py`, and all MTGJSON tests/fixtures
- Migrated all consumers (card_resolver, analysis workflow, orchestrator) to Scryfall bulk data
- Config: `enable_mtgjson` → `enable_bulk_data`, `mtgjson_*` settings removed

### New: Tiered test infrastructure
- `tests/integration/` — fixture-mocked cross-component E2E tests (bulk pipeline + orchestrator)
- `tests/live/` — real server subprocess + real API smoke tests (`@pytest.mark.live`)
- `mise run check:quick` — fast quality gate (lint + typecheck + testmon)
- CI: PRs run `check` + `live-tests` + `build` + `security` in parallel

### Docs
- Phase 6 design specs: `SPEC_INFRASTRUCTURE_MODERNIZATION.md`, `SPEC_CROSS_FORMAT_TOOLS.md`, `SPEC_FORMAT_WORKFLOWS.md`
- ARCHITECTURE.md fully updated for MTGJSON → bulk data migration
- CLAUDE.md updated with testing strategy and new commands

## Test plan

- [x] `mise run check` — 479 tests pass, lint + typecheck clean
- [x] `mise run test:live` — live smoke tests pass against real Scryfall data
- [x] `mise run test:integration` — integration tests pass
- [x] Interactive smoke test via `/smoke-test` — all 15 tool categories verified

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by j4th** [comment]
@claude Comprehensive review of this PR, and assessment of results alignments with our original spec.

File: tests/live/conftest.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import httpx
+import pytest
+from fastmcp import Client
+
+
+def _find_free_port() -> int:
+    """Bind to port 0 and let the OS assign a free port."""
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("", 0))
```
[the remark this answers is not in the document]
Comment (j4th) [reply tests/live/conftest.py]: Stale finding — this was fixed in 671af52 to bind to `"127.0.0.1"`. The scanner is reporting against an old diff.

**Comment by j4th** [comment]
## Review response

Addressed 5 of 7 items from the review:

| # | Item | Action |
|---|------|--------|
| 1 | Security scanner (stale) | **Skipped** — already fixed in 671af52, replied to dismiss |
| 2 | Hardcoded `BULK_DOWNLOAD_URL` | **Fixed** — now derived from `_load_bulk_metadata()["download_uri"]` |
| 3 | Bare `except Exception` in card_resolver | **Deferred** — intentional fallback with `exc_info=True` logging; narrowing risks missing bulk client edge cases |
| 4 | Empty string append in analysis.py | **Fixed** — replaced ternary with `if deck_size > 0` guard |
| 5 | Magic number 23 in tests | **Fixed** — added `# 1 ping + 4 scryfall + 4 spellbook + 2 draft + 2 edhrec + 2 bulk + 8 workflows = 23` |
| 6 | Lifespan assignment order | **Fixed** — `_client = client` now before `start_background_refresh()` |
| 7 | Timeout on base test task | **Fixed** — `mise run test` now includes `--timeout=300` |

All 479 tests pass, 94% coverage, lint + typecheck clean.
